unitTests/ui/MissingTermsTest.ts

```diff
@@ -0,0 +1,146 @@
+import * as Mock from '../MockEnvironment';
+import { MissingTerms } from '../../src/ui/MissingTerm/MissingTerms';
+import { $$ } from '../../src/Core';
+import { FakeResults } from '../Fake';
+
+export function MissingTermsTest() {
+  describe('MissingTerm', () => {
+    let test: Mock.IBasicComponentSetup<MissingTerms>;
+    let fakeResult = FakeResults.createFakeResult();
+    fakeResult.state.q = 'This will be a really really really long query';
+    const wordNotInQuery = 'foo';
+    const wordPresent = 'will';
+    const firstWord = 'This';
+    const getSpy: any = jasmine.createSpy('getSpy').and.returnValue('This will be a really really really long query');
+    beforeEach(() => {
+      fakeResult.absentTerms = [];
+      fakeResult.absentTerms.push(wordPresent);
+    });
+
+    describe('exposes options', () => {
+      describe("when the 'clickable' option is set to", () => {
+        let missingTermsClickableSpy: jasmine.Spy;
+        const clickOnMissingTerms = () => {
+          test.cmp.includeTermInQuery = missingTermsClickableSpy;
+          const missingTermElement = $$(test.cmp.element).find('.coveo-missing-term');
+          missingTermElement.click();
+        };
+
+        beforeEach(() => {
+          missingTermsClickableSpy = jasmine.createSpy('MissingTermsClickableSpy');
+        });
+
+        it('true should allow the user to click on the missingTerm', () => {
+          test = Mock.advancedResultComponentSetup<MissingTerms>(
```

Comment:
There's a lot of repetition in your test setup, with only some option value changing between each, or minor difference.

It should be easy to extract some easy to re-use setup function if you spend some time polishing. 

The tests are pretty hard to read as they stand right now, so I'll stop my review here.

You can create setup function/utils function to remove most of the repetitive code in this file.